src/lib.rs

```diff
@@ -89,55 +91,36 @@ impl Contract {
         }
     }
 
-    pub fn claim(&mut self) -> PromiseOrValue<WrappedBalance> {
+    pub fn claim_lockups(
```

Comment:
Let's think about the public interface a bit more. I don't like `claim` + `claim_lockups` API since it is not clear how those are different.

I would go with a single `claim` method and just define the `amounts` argument as optional, so it won't be a breaking change, and it will also have a cleaner API.